lib/dialects/mysql/connection-manager.js

```diff
@@ -69,82 +85,93 @@ class ConnectionManager extends AbstractConnectionManager {
         }
       }
 
-      const connection = this.lib.createConnection(connectionConfig);
-
-      connection.connect(err => {
-        if (err) {
-          if (err.code) {
-            switch (err.code) {
-            case 'ECONNREFUSED':
-              reject(new sequelizeErrors.ConnectionRefusedError(err));
-              break;
-            case 'ER_ACCESS_DENIED_ERROR':
-              reject(new sequelizeErrors.AccessDeniedError(err));
-              break;
-            case 'ENOTFOUND':
-              reject(new sequelizeErrors.HostNotFoundError(err));
-              break;
-            case 'EHOSTUNREACH':
-              reject(new sequelizeErrors.HostNotReachableError(err));
-              break;
-            case 'EINVAL':
-              reject(new sequelizeErrors.InvalidConnectionError(err));
-              break;
-            default:
-              reject(new sequelizeErrors.ConnectionError(err));
-              break;
-            }
-          } else {
-            reject(new sequelizeErrors.ConnectionError(err));
-          }
-
-          return;
-        }
+      return new Utils.Promise((resolve, reject) => {
+        const connection = this.lib.createConnection(connectionConfig);
+        connection.once('error', (err) => {
+          reject(err);
+        });
+        connection.once('connect', () => {
+          connection.removeAllListeners('error');
```

Comment:
@janmeier Connection lost error are handled in next `.then` block. https://github.com/sequelize/sequelize/pull/6354/files#diff-ed5bbd2edaf04e2052f0919e5658fe22R107

@felixfbecker If we don't remove the `error`handler in connect that we will have two erro events attached [1) reject](https://github.com/sequelize/sequelize/pull/6354/files#diff-ed5bbd2edaf04e2052f0919e5658fe22R90) and [2) lost connection handler](https://github.com/sequelize/sequelize/pull/6354/files#diff-ed5bbd2edaf04e2052f0919e5658fe22R107), where we only need (2). So its required to removed (1) in `connect` phase